kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,307 @@
+    return getConsumerGroups(adminClientWrapper.listConsumerGroups());
+  }
+
+  private List<ConsumerGroup> getConsumerGroups(Collection<ConsumerGroupListing> groupsOverview)
+      throws Exception {
+    final List<ConsumerGroup> result = new ArrayList<>();
+    for (ConsumerGroupListing eachGroupInfo : groupsOverview) {
+      final Node node = adminClientWrapper.describeConsumerGroups(
+           Collections.singleton(eachGroupInfo.groupId()))
+               .get(eachGroupInfo.groupId()).coordinator();
+      result.add(new ConsumerGroup(eachGroupInfo.groupId(),
+          new ConsumerGroupCoordinator(node.host(), node.port())));
+    }
+    return result;
+  }
+
+  private Collection<ConsumerGroupListing> getPagedConsumerGroup(Integer startPosition,
+                                                                 Integer count)
+      throws Exception {
+    Collection<ConsumerGroupListing> groupsOverview;
+    final List<ConsumerGroupListing> consumerGroupListings =
+        new ArrayList<>(adminClientWrapper.listConsumerGroups());
+    consumerGroupListings.sort(Comparator.comparing(ConsumerGroupListing::groupId));
+    groupsOverview = consumerGroupListings.subList(startPosition,
+        Math.min(consumerGroupListings.size(), startPosition + count));
+    return groupsOverview;
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   */
+  public Set<Topic> getConsumerGroupTopicInformation(String groupId)
+          throws Exception {
+    final List<Topic> result = getConsumerGroupTopics(groupId);
+    log.debug("Get topic list {}", result);
+    return new HashSet<>(result);
+  }
+
+  /**
+   * <p>Get consumer group description restricted by paging parameters</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   */
+  public Set<Topic> getPagedConsumerGroupTopicInformation(String groupId,
+                                                          Integer startPos,
+                                                          Integer count)
+          throws Exception {
+    final List<Topic> result = getConsumerGroupTopics(groupId);
+    log.debug("Get topic list {}", result);
+    return new HashSet<>(result.subList(startPos,
+        Math.min(result.size(), startPos + count)));
+  }
+
+  private List<Topic> getConsumerGroupTopics(String groupId) throws Exception {
+    final List<Topic> result = new ArrayList<>();
+    final Collection<MemberDescription> memberDescriptions =
+        adminClientWrapper.describeConsumerGroups(Collections.singleton(groupId))
+            .get(groupId).members();
+    if (memberDescriptions.isEmpty()) {
+      return Collections.emptyList();
+    }
+    for (MemberDescription eachSummary : memberDescriptions) {
+      for (TopicPartition topicPartition : eachSummary.assignment().topicPartitions()) {
+        result.add(new Topic(topicPartition.topic(), null, null));
```

Comment:
Given that we only care about the topic name, why not return a Set<String> here and Set<String> on the public methods?